subt_ign/src/GameLogicPlugin.cc

```diff
@@ -1012,6 +1024,60 @@ void GameLogicPluginPrivate::OnEvent(const ignition::msgs::Pose &_msg)
   }
 }
 
+//////////////////////////////////////////////////
+void GameLogicPlugin::PreUpdate(const UpdateInfo &_info,
+                                 EntityComponentManager &_ecm)
+{
+  if (_info.dt < std::chrono::steady_clock::duration::zero())
+  {
+    ignwarn << "Detected jump back in time ["
+        << std::chrono::duration_cast<std::chrono::seconds>(_info.dt).count()
+        << "s]. System may not work properly." << std::endl;
+  }
+
+  if (_info.paused)
+    return;
+
+  if (!this->dataPtr->started)
+  {
+    _ecm.Each<gazebo::components::Sensor,
+              gazebo::components::ParentEntity>(
+        [&](const gazebo::Entity &,
+            const gazebo::components::Sensor *,
+            const gazebo::components::ParentEntity *_parent) -> bool
+        {
+          // Get the model. We are assuming that a sensor is attached to
+          // a link.
+          auto model = _ecm.Component<gazebo::components::ParentEntity>(
+              _parent->Data());
+
+          if (model)
+          {
+            if (!_ecm.Component<components::HaltMotion>(model->Data()))
+            {
+              _ecm.CreateComponent(model->Data(),components::HaltMotion(false));
+            }
+          }
+          return true;
+        });
+  }
+  else
+  {
+    for (std::string robotName : this->dataPtr->haltedRobots)
+    {
+      if (this->dataPtr->robotEntityMap.find(robotName) ==
+          this->dataPtr->robotEntityMap.end())
+          return;
```

Comment:
```suggestion
          continue;
```

Reply:
Agree, but I think a case could be made about why do you have a halted robot that is not inside the Entity map